tests/bluetooth/host/host_mocks/print_utils.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include "host_mocks/print_utils.h"
+
+static int bt_addr_le_to_str_mod(const bt_addr_le_t *addr, char *str,
+					size_t len)
+{
+	char type[10];
+
+	switch (addr->type) {
+	case BT_ADDR_LE_PUBLIC:
+		strcpy(type, "public");
+		break;
+	case BT_ADDR_LE_RANDOM:
+		strcpy(type, "random");
+		break;
+	case BT_ADDR_LE_PUBLIC_ID:
+		strcpy(type, "public-id");
+		break;
+	case BT_ADDR_LE_RANDOM_ID:
+		strcpy(type, "random-id");
+		break;
+	default:
+		snprintf(type, sizeof(type), "0x%02x", addr->type);
+		break;
+	}
+
+	return snprintf(str, len, "%02X:%02X:%02X:%02X:%02X:%02X (%s)",
+			addr->a.val[5], addr->a.val[4], addr->a.val[3],
+			addr->a.val[2], addr->a.val[1], addr->a.val[0], type);
+}
+
+static int bt_addr_to_str_mod(const bt_addr_t *addr, char *str, size_t len)
+{
+	return snprintf(str, len, "%02X:%02X:%02X:%02X:%02X:%02X",
+			addr->val[5], addr->val[4], addr->val[3],
+			addr->val[2], addr->val[1], addr->val[0]);
+}
+
+const char *bt_addr_str_real(const bt_addr_t *addr)
+{
+	static char str[BT_ADDR_STR_LEN];
+
+	bt_addr_to_str_mod(addr, str, sizeof(str));
+
+	return str;
+}
+
+const char *bt_addr_le_str_real(const bt_addr_le_t *addr)
+{
+	static char str[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str_mod(addr, str, sizeof(str));
+
+	return str;
+}
+
+const char *bt_hex_real(const void *buf, size_t len)
```

Comment:
why can't we use `bluetooth/common/log.c`? Also, where are you using this, is it for visualization or error messages in the UTs?

Reply:
Ok, `bluetooth/host/keys.c` uses `BT_DBG(fmt, ...)` which brings `log` module definitions directly.

Firstly, I tried to add  `bluetooth/common/log.c`, but there were a lot of other dependencies and compilations issues that I preferred to avoid as I'm not really interested in 'log.c' itself.

Regarding the UTs, these functions are just used to print a readable strings when some asserts fail.